include/crow/middlewares/cookie_parser.h

```diff
@@ -43,14 +166,17 @@ namespace crow
                 return {};
             }
 
-            void set_cookie(const std::string& key, const std::string& value)
+            template<typename U>
+            Cookie& set_cookie(const std::string& key, U&& value)
             {
-                cookies_to_add.emplace(key, value);
+                cookies_to_add.emplace_back(key, std::forward<U>(value));
```

Comment:
Might be a stupid question but why use a template instead of just a string? I'm guessing it's to be able to put things like integers, but I'm not sure.

Reply:
The cookie constructor is templatized as well.
This allows to create cookies without copying the value in many cases:
- move a string
- pass a string_view, const char*
- pass anything that is assignable to a string
This doesn't make sense for keys, as they are very short. Values can be up to a few kb
Arbitrary values (like integers) are not supported.

Reply:
Thank you for clarifying!